contributions/api/utils/query_params.py

```diff
@@ -14,39 +14,48 @@
 
 from bson import ObjectId
 
-def format_query_contribution(args, query):
+def format_query_contribution(args, query, is_args=False):
     query_parts = []
 
-    if args.get('id'):
-        # make sure it has been changed to eventIds from eventid since the db field name is eventIds
-        query_parts.append({'_id': {'$eq': ObjectId(args.get('id'))}})
-    if args.get('name'):
-        query_parts.append({'name': {'$eq': args.get('name')}})
+    if is_args:
+        if args.get('name'):
+            query_parts.append({'name': {'$eq': args.get('name')}})
+    else:
+        if args is not None:
+            query_parts.append({'name': args})
```

Comment:
when is_arg = false and args = None, so the query returns directly? I am trying to figuring out the relationship between args and is_args...

Reply:
In that case, this method is probably not being called. If you see from api/controllers/contributions.py's line 133, the case of args is none is treated as whole list, so the above method is not actually being called. I put the codes like above to avoid the exception raise that causes the application being stopped.... like an insurance.

Reply:
So args is the variable of the actual argument, and is_args is a boolean to see if it is using argument or input query variable that is 'name' in this case

Reply:
this will be called only when `name` presented...ok i see!